### PR TITLE
ServiceAccount and TerminationGracePeriod

### LINKED ISSUE #10: TerminationGracePeriod Support
Currently termination grace period is set to default ie 30 sec. Can we add support in the spec so that we can modify the grace period.

### LINKED ISSUE #24: Deploy druid cluster in Service Account and support ImagePullSecrets
I propose to deploy the druid cluster in a service account so that adequate RBAC's can be applied in a prod environment. 
Currently operator cannot pull custom built images private registeries which require authentication, add support for ImagePullSecrets option in sts.
Currently only the operator runs in a SA. Not sure  adding SA in druid cluster spec should it be a Required or Optional Parameter ? 
@himanshug could you suggest on service account

### PR DESCRIPTION
**Pull request by AdheipSingh**
Fixes #10 
Fixes #24 

### Description

TerminationGrace Period is limited to 30sec as default. Some processes such as middle managers, historicals needs more graceful time during an update.
ServiceAccount, deploys the cluster in a specified SA, helps in giving access, applying RBACs to druid cluster across a cluster

- This func returns the TGP if defined else keeps is too 30. When directly using TGP without the func k8s keeps the sts TGP to 0. 
```
func getTerminationGracePeriod(nodeSpec *v1alpha1.DruidNodeSpec) 
```
<hr>

This PR has:
- [x ] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [ x] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * ```handler.go```
 * ```druid_types.go```

### CODE REVIEW

File: pkg/controller/druid/handler.go [hunk 1]
```diff
@@ -675,6 +675,15 @@ func updateDefaultPortInProbe(probe *v1.Probe, defaultPort int32) *v1.Probe {
 	return probe
 }
 
+func getTerminationGracePeriod(nodeSpec *v1alpha1.DruidNodeSpec) *int64 {
+
+	if nodeSpec.TerminationGracePeriodSeconds == 0 {
+		nodeSpec.TerminationGracePeriodSeconds = 30
```
Comment (himanshug) [inline pkg/controller/druid/handler.go]: can we leave it be default value that k8s cluster would have instead of hardcoding 30 secs in the operator

Comment (AdheipSingh) [reply pkg/controller/druid/handler.go]: I had kept to absolute value, changed type to pointer.

File: pkg/apis/druid/v1alpha1/druid_types.go [hunk 2]
```diff
@@ -157,6 +161,9 @@ type DruidNodeSpec struct {
 	// Optional: affinity to be used to for enabling node, pod affinity and anti-affinity
 	Affinity *v1.Affinity `json:"affinity,omitempty"`
 
+	// Optional: terminationGracePeriod defaults to 30 sec.
```
Comment (himanshug) [inline pkg/apis/druid/v1alpha1/druid_types.go]:
```suggestion
	// Optional: terminationGracePeriod
```

Comment (AdheipSingh) [reply pkg/apis/druid/v1alpha1/druid_types.go]: Done

**Pull request merged**